### PR TITLE
Update envoy and contour

### PR DESCRIPTION
**Pull request by chez-shanpu**

### CODE REVIEW

**Comment by chez-shanpu** [comment]
depends on https://github.com/cybozu/neco-containers/pull/1224

File: envoy/go.mod [hunk 1]
```diff
@@ -1,33 +1,39 @@
 module github.com/cybozu/neco-containers/envoy
 
-go 1.20
+go 1.21
+
+toolchain go1.21.4
```
Comment (YZ775) [inline envoy/go.mod:5]: Is this directive required?

**Comment by YZ775** [comment]
I think we need to update go version in envoy's Dockerfile

https://github.com/cybozu/neco-containers/blob/b664e5ccd3283f0f89574dcd863b183a9b100674/envoy/Dockerfile#L4

File: envoy/go.mod (hunk 1 above)
Comment (chez-shanpu) [reply envoy/go.mod:5]: It wouldn't be required. I think it was added somehow when I update its dependencies. I'll remove it :)

**Review by YZ775 — APPROVED** [review]
LGTM

**Pull request merged**